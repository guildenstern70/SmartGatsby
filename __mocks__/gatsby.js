/*
 * Project SmartGatsby
 * Copyright (c) Alessio Saltarin 2021
 * Licensed under MIT license
 */

/* eslint-disable no-unused-vars */
const React = require("react");
const gatsby = jest.requireActual("gatsby");

module.exports = {
	...gatsby,
	graphql: jest.fn(),
	Link: jest.fn().mockImplementation(
		// these props are invalid for an `a` tag
		({
			 activeClassName,
			 activeStyle,
			 getProps,
			 innerRef,
			 partiallyActive,
			 ref,
			 replace,
			 to,
			 ...rest
		 }) =>
			React.createElement("a", {
				...rest,
				href: to,
			})
	),
	StaticQuery: jest.fn(),
	useStaticQuery: jest.fn(),
}
