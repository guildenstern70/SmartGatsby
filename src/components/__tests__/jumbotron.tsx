/*
 * Project SmartGatsby
 * Copyright (c) Alessio Saltarin 2021
 * Licensed under MIT license
 */


import * as React from "react";
import * as renderer from "react-test-renderer";
import Jumbotron from "../jumbotron";
import { useStaticQuery } from "gatsby";

beforeEach(() => {
	// @ts-ignore
	useStaticQuery.mockImplementation( ()  =>
		({
			"site": {
				"siteMetadata": {
					"title": "SmartGatsby",
				},
			},
		})
	)});


it("renders correctly", () => {

  const tree = renderer
    .create(
      <Jumbotron subtitle={"Built with Gatsby"}
                 description={"A Gatsby template that makes use of Bootstrap 5 and TypeScript"}
                 linkUrl={"#"} linkTitle={"Login"} />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
