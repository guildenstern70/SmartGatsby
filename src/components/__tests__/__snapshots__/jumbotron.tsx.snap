// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="jumbotron"
>
  <h2
    className="display-5 jumbotitle text-primary"
  >
    SmartGatsby
  </h2>
  <p
    className="lead"
  >
    Built with Gatsby
  </p>
  <hr
    className="my-4"
  />
  <p>
    A Gatsby template that makes use of Bootstrap 5 and TypeScript
  </p>
  <a
    className="btn btn-primary btn-lg"
    href="#"
    role="button"
  >
    Login
  </a>
</div>
`;
